### PR TITLE
Utilize newly implemented functions in ce-dataprep

### LINKED ISSUE #68: Variable split_intervals is never used in wps_generate_climos
`wps_generate_climos` takes an input [split_intervals](https://github.com/pacificclimate/thunderbird/blob/master/thunderbird/processes/wps_generate_climos.py#L190) but is never used when `thunderbird.generate_climos` is called. I believe [generate_climos](https://github.com/pacificclimate/climate-explorer-data-prep#usage) use this option for splitting the output files depending on time intervals. If this option will not be used in thunderbird, I think it should be removed from the input options.

### LINKED ISSUE #76: Modification is needed for wps_generate_climos due to the most recent release of ce-dataprep 0.8.4
cd-dataprep 0.8.4 version has been released. Now preprocesses of `wps_generate_climo` to run `create_climo_files` can be reduced by using `dry_run_handler` and `generate_climos`

### PR DESCRIPTION
**Pull request by sum1lim**
This PR closes #76 

`wps_genereate_climos` now uses the same strategy of `generate_climos` script of `ce-dataprep`.
`dry_run_handler` and `generate_climos` functions are imported from `dp` package and replace the functionally equivalent codes in `self._handler`.

There was an issue that the previous version of `ce-dataprep` did not pass `split_intervals` argument into `create_climo_files`. It has been resolved in the latest version 0.8.4 by implementing `generate_climos` function. Therefore, calling the function also resolves #68

For dry_run, logging messages of `dry_run_handler` are redirected to the output `.txt` file, and an example looks like below:
```
Dry Run
generate_climos:
INFO:dp.generate_climos:Processing: https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/dodsC/datasets/TestData/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100.nc
INFO:dp.generate_climos:climo_periods: {'6190'}
INFO:dp.generate_climos:project: CMIP5
INFO:dp.generate_climos:institution: PCIC
INFO:dp.generate_climos:model: CanESM2
INFO:dp.generate_climos:emissions: historical, rcp85
INFO:dp.generate_climos:run: r1i1p1
INFO:dp.generate_climos:dependent_varnames: ['fdd']
INFO:dp.generate_climos:time_resolution: seasonal
INFO:dp.generate_climos:is_multi_year_mean: False
```

### CODE REVIEW

File: thunderbird/processes/wps_generate_climos.py [hunk 1]
```diff
@@ -139,7 +139,8 @@ def dry_run_info(self, filepath, climo):
             f.write("Dry Run\n")
             logging.basicConfig(filename=filename, level=logging.INFO)
             dry_run_handler(filepath, climo)
-            logging.basicConfig(stream=sys.stderr, level=logging.INFO)
```
Comment (eyvorchuk) [inline thunderbird/processes/wps_generate_climos.py]: Since you removed this statement, you can remove `import sys` at the top.

Comment (sum1lim) [reply thunderbird/processes/wps_generate_climos.py]: Good point! I totally forgot about that.

File: thunderbird/processes/wps_generate_climos.py [hunk 2]
```diff
@@ -129,32 +130,17 @@ def __init__(self):
...
         filename += "_dry.txt"
         filename = os.path.join(self.workdir, filename)
         with open(filename, "w") as f:
-            for line in output:
-                f.write(f"{line}\n")
+            f.write("Dry Run\n")
+            logging.basicConfig(filename=filename, level=logging.INFO)
+            dry_run_handler(filepath, climo)
+            for handler in logging.root.handlers[:]:
+                logging.root.removeHandler(handler)
```
Comment (nikola-rados) [inline thunderbird/processes/wps_generate_climos.py:142]: So as I understand it, we are logging to the `dry.txt` file here through the `python logging` module. I think this is fairly messy and could be done in a better way. In fact, we do this in a better way in the [`generate_prsn` module](https://github.com/pacificclimate/thunderbird/blob/master/thunderbird/processes/wps_generate_prsn.py#L153). We use a [`dry` method](https://github.com/pacificclimate/climate-explorer-data-prep/blob/master/dp/generate_prsn.py#L22) that has a flexible output. I think we need to double back to `ce-dataprep` and adjust the `dry` method. It was an oversight on my part not to see the issue ahead of time, my bad.

Here is the new [issue](https://github.com/pacificclimate/climate-explorer-data-prep/issues/135).

File: thunderbird/processes/wps_generate_climos.py [hunk 3]
```diff
@@ -228,29 +214,24 @@ def _handler(self, request, response):
...
+                generate_climos(
+                    filepath,
+                    self.workdir,
+                    operation,
+                    climo,
+                    convert_longitudes=convert_longitudes,
+                    split_vars=split_vars,
+                    split_intervals=split_intervals,
+                    resolutions=resolutions,
+                )
```
Comment (nikola-rados) [inline thunderbird/processes/wps_generate_climos.py:233]: Great job here, this has cleaned things up even more than before.

File: thunderbird/processes/wps_generate_climos.py [hunk 4]
```diff
@@ -129,32 +130,17 @@ def __init__(self):
...
-            output.append(f"climo_periods: {periods}")
-            for attr in "project institution model emissions run".split():
-                try:
-                    output.append(f"{attr}: {getattr(input_file.metadata, attr)}")
-                except Exception as e:
-                    output.append(f"{attr}: {e.__class__.__name__}: {e}")
-            output.append(f"dependent_varnames: {input_file.dependent_varnames()}")
-            for attr in "time_resolution is_multi_year_mean".split():
-                output.append(f"{attr}: {getattr(input_file, attr)}")
-
```
Comment (nikola-rados) [inline thunderbird/processes/wps_generate_climos.py:149]: Nice to remove this repetitive code!

**Comment by sum1lim** [comment]
I have ensured the dry_run logging messages are successfully written to output files by running demos.
For example, running the following:
```
# Test multiple files
dry_output = thunderbird.generate_climos(
    netcdf=[seasonal_opendap, annual_local], 
    operation=operation, 
    climo=climo, 
    resolutions=resolutions, 
    dry_run=dry_run
)
```
will create two output files:
```
[<metaurl mediatype="text/plain">
      http://localhost:5001/outputs/6cc098bf-cb9b-11ea-9e56-503eaa25cee0/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100_dry.txt
     </metaurl>,
 <metaurl mediatype="text/plain">
      http://localhost:5001/outputs/6cc098c0-cb9b-11ea-9e56-503eaa25cee0/gdd_annual_CanESM2_rcp85_r1i1p1_1951-2100_dry.txt
     </metaurl>]
```
The contents are:
```
Dry Run
generate_climos:
INFO:dp.generate_climos:Processing: /tmp/pywps_process_g3nhrxhx/gdd_annual_CanESM2_rcp85_r1i1p1_1951-2100.nc
INFO:dp.generate_climos:climo_periods: {'6190'}
INFO:dp.generate_climos:project: CMIP5
INFO:dp.generate_climos:institution: PCIC
INFO:dp.generate_climos:model: CanESM2
INFO:dp.generate_climos:emissions: rcp85
INFO:dp.generate_climos:run: r1i1p1
INFO:dp.generate_climos:dependent_varnames: ['gdd']
INFO:dp.generate_climos:time_resolution: yearly
INFO:dp.generate_climos:is_multi_year_mean: False
```
```
Dry Run
generate_climos:
INFO:dp.generate_climos:Processing: https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/dodsC/datasets/TestData/fdd_seasonal_CanESM2_rcp85_r1i1p1_1951-2100.nc
INFO:dp.generate_climos:climo_periods: {'6190'}
INFO:dp.generate_climos:project: CMIP5
INFO:dp.generate_climos:institution: PCIC
INFO:dp.generate_climos:model: CanESM2
INFO:dp.generate_climos:emissions: historical, rcp85
INFO:dp.generate_climos:run: r1i1p1
INFO:dp.generate_climos:dependent_varnames: ['fdd']
INFO:dp.generate_climos:time_resolution: seasonal
INFO:dp.generate_climos:is_multi_year_mean: False
```

File: requirements.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-ce-dataprep==0.8.3
+ce-dataprep==0.8.4
```
Comment (nikola-rados) [inline requirements.txt:1]: Bump this.

File: thunderbird/processes/wps_generate_climos.py [hunk 6]
```diff
@@ -129,32 +129,17 @@ def __init__(self):
...
         filename += "_dry.txt"
         filename = os.path.join(self.workdir, filename)
         with open(filename, "w") as f:
-            for line in output:
-                f.write(f"{line}\n")
+            f.write("Dry Run\n")
+            logging.basicConfig(filename=filename, level=logging.INFO)
+            dry_run_handler(filepath, climo)
+            for handler in logging.root.handlers[:]:
+                logging.root.removeHandler(handler)
```
Comment (nikola-rados) [inline thunderbird/processes/wps_generate_climos.py:142]: It would be helpful to put a comment in here explaining what is happening.

**Pull request merged**